app/gui2/src/stores/graph/index.ts

```diff
@@ -312,16 +312,24 @@ export const { injectFn: useGraphStore, provideFn: provideGraphStore } = createC
 
     function deleteNodes(ids: Iterable<NodeId>) {
       edit((edit) => {
+        const deletedNodes = new Set()
```

Comment:
I think we should collect all `ids` ahead of the main loop avoid modifying nodes that we are about to delete as well.